src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,456 @@
+	const unsigned int hdd_seek_volume = prop->Get_int("hdd_seek_volume");
+
+	std::vector<std::string> floppy_seek_paths;
+	std::vector<std::string> hdd_seek_paths;
+
+	for (int i = 1; i <= 9; ++i) {
+		floppy_seek_paths.push_back(prop->Get_string(
+		        "floppy_seek_sample_" + std::to_string(i)));
+		hdd_seek_paths.push_back(
+		        prop->Get_string("hdd_seek_sample_" + std::to_string(i)));
+	}
+
+	if (!floppy_spin.empty() ||
+	    !std::all_of(floppy_seek_paths.begin(),
+	                 floppy_seek_paths.end(),
+	                 [](const auto& s) { return s.empty(); })) {
+		floppy_noise = std::make_unique<DiskNoiseDevice>(enable_disk_noise,
+		                                                 floppy_spin_up,
+		                                                 floppy_spin,
+		                                                 floppy_seek_paths,
+		                                                 "FLOPPY_SPIN",
+		                                                 "FLOPPY_SEEK",
+		                                                 floppy_spin_volume,
+		                                                 floppy_seek_volume);
+	}
+
+	if (!hdd_spin.empty() ||
+	    !std::all_of(hdd_seek_paths.begin(),
+	                 hdd_seek_paths.end(),
+	                 [](const auto& s) { return s.empty(); })) {
+		hdd_noise = std::make_unique<DiskNoiseDevice>(enable_disk_noise,
+		                                              hdd_spin_up,
+		                                              hdd_spin,
+		                                              hdd_seek_paths,
+		                                              "HDD_SPIN",
+		                                              "HDD_SEEK",
+		                                              hdd_spin_volume,
+		                                              hdd_seek_volume);
+	}
+}
+
+void DISKNOISE_ShutDown([[maybe_unused]] Section* section)
+{
+	floppy_noise->Shutdown();
+	hdd_noise->Shutdown();
+}
+
+void init_disknoise_dosbox_settings(Section_prop& secprop)
+{
+	constexpr auto only_at_start = Property::Changeable::OnlyAtStart;
+
+	auto* bool_prop = secprop.Add_bool("enable_disk_noise_emulation",
+	                                   only_at_start,
+	                                   false);
+	bool_prop->Set_help(
+	        "Enables emulated disk noises for hard disks and floppy drives.\n"
+	        "This plays disk spinning and seek noises using configurable samples.\n"
+	        "Up to 9 seek samples are supported per disk type. Sample 1 has\n"
+	        "the highest chance of being played, and sample 9 has the lowest chance.\n"
+	        "Sample should be in 16-bit PCM WAV format at 44100 Hz, and can be\n"
+	        "located in the dosbox resources directory or beconfigured with an\n"
+	        "absolute path.\n"
+	        "This can be combined with the 'hdd_io_speed' and 'fdd_io_speed settings'\n"
+	        "for an authentic experience. This feature is disabled by default.\n");
```

Comment:
Nice job documenting this thoroughly, I appreciate that 👍🏻 However, as discussed, it's too much. Let's start with a simple `on` and `off` option.